render/src/bitmap.rs

```diff
@@ -9,12 +11,19 @@ pub struct BitmapInfo {
     pub height: u16,
 }
 
+#[derive(Copy, Clone, Debug)]
+pub struct BitmapSourceInfo {
```

Comment:
Minor naming nit: Instead of `BitmapSourceInfo` this probably should just be `BitmapSize`, since it's just a width and height.